python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -180,6 +180,10 @@ private void createAmbiguousSymbols() {
         case FUNC_DECLARATION:
           FunctionDef functionDef = (FunctionDef) bindingUsage.tree().parent();
           FunctionSymbolImpl functionSymbol = new FunctionSymbolImpl(functionDef, symbol.fullyQualifiedName(), pythonFile);
+          ParameterList parameters = functionDef.parameters();
```

Comment:
Why can't we do it inside `FunctionSymbolImpl` constructor anymore?

Reply:
I tried adding that logic again in `FunctionSymbolImpl` constructor but encountered a different issue where some class symbols could have their super class read through type annotations before their type hierarchy was defined.
Therefore, in d9e48824b4d75f149e2db5cfcac09b46ad1a8fe9, I moved this logic to `ThirdPhaseVisitor`.